Fix Bazel remote cache unreachable inside Docker containers (Connection refused: 127.0.0.1:9090)

## Problem

The Bazel remote cache at `http://rayci.localhost:9090` is unreachable from inside forge Docker containers, causing thousands of cache miss warnings and slower builds.

In build 254, the reef test step (g3_s0) logs show:
```
WARNING: Remote Cache: Connection refused: /127.0.0.1:9090
io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: /127.0.0.1:9090
Caused by: java.net.ConnectException: Connection refused
```

### Root Cause

The `BUILDKITE_BAZEL_CACHE_URL` in `.buildkite/fork-config.yaml` is set to `http://rayci.localhost:9090`. The `.localhost` TLD resolves to `127.0.0.1` per RFC 6761. On the **host**, this correctly reaches the cache server on port 9090. But inside a **Docker container**, `127.0.0.1` refers to the container's own loopback, not the host — so the connection fails.

### Impact

- **Not a blocker**: Bazel builds still succeed — they just skip the remote cache and build from source every time
- **Performance**: Every build step inside a Docker container does a full uncached build, significantly increasing build times
- **Affected steps**: All test steps that run inside forge Docker containers (reef, core tests, etc.)

### Fix

Add `add-host` entries to the `docker_plugin` section of `.buildkite/fork-config.yaml` to map `rayci.localhost` to the host's IP address inside Docker containers:

```yaml
docker_plugin:
  allow_mount_buildkite_agent: true
  add-host:
    - "rayci.localhost:host-gateway"
```

The `host-gateway` special value is supported by Docker 20.10+ and maps to the host machine's gateway IP (typically `172.17.0.1` on the default bridge network).

**Important**: Before implementing, verify that rayci's `docker_plugin` config section passes `add-host` entries through to the Docker Buildkite plugin. Check the generated `pipeline_flat.yaml` artifact to confirm the `add-host` directive appears in the Docker plugin configuration.

### Alternative Approaches (if `add-host` is not supported by rayci)

1. **Change the URL to use host.docker.internal**: Set `BUILDKITE_BAZEL_CACHE_URL: "http://host.docker.internal:9090"` — but this requires `--add-host=host.docker.internal:host-gateway` which may or may not be the default on the build agent.

2. **Use the host's actual LAN IP**: If the cache server binds to all interfaces, use the host's actual IP address (e.g., `192.168.x.x`). This avoids the loopback resolution issue but makes the config host-specific.

3. **Disable the cache**: Set `BUILDKITE_BAZEL_CACHE_URL: ""` to suppress the warnings. Not ideal since it gives up on caching entirely.

### Verification

After merging, trigger a new build and check:
1. No `Connection refused: /127.0.0.1:9090` errors in reef or core test step logs
2. Bazel output shows cache hits (look for `remote cache hit` in build output)
3. Build step durations decrease compared to build 254

### Files to Modify

- `.buildkite/fork-config.yaml` — Add `add-host` to `docker_plugin` section

### Evidence

Build 254 logs (step `019ce610-0db8-4083-99e4-07cb3455021b`):
- Lines 948, 1228+: `WARNING: Remote Cache: Connection refused: /127.0.0.1:9090`
- Multiple bulk transfer errors with full Java stack traces

--- a/.buildkite/fork-config.yaml
+++ b/.buildkite/fork-config.yaml
@@ -57,6 +57,8 @@ hook_env_keys:
 
 docker_plugin:
   allow_mount_buildkite_agent: true
+  add-host:
+    - "rayci.localhost:host-gateway"
 
 # Tags that cause steps to be unconditionally skipped.
 skip_tags:

